src/IO/ResourceInputStream.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace PhpSchool\CliMenu\IO;
+
+use function is_resource;
+use function get_resource_type;
+use function stream_get_meta_data;
+use function strpos;
+
+/**
+ * @author Aydin Hassan <aydin@hotmail.co.uk>
+ */
+class ResourceInputStream implements InputStream
+{
+    /**
+     * @var resource
+     */
+    private $stream;
+
+    public function __construct($stream = null)
+    {
+        if ($stream === null) {
```

Comment:
maybe I'm being silly, can't you have `STDIN` as the default value?

Reply:
Let's give it a whirl

Reply:
Indeed we can!